File: backend/internal/model/calculate.go
package model

type FloorCalculationRequest struct {
	FloorType      string  `json:"floor_type"`
	Area           float64 `json:"area"`
	RemoveOldFloor bool    `json:"remove_old_floor"`
	Environment    string  `json:"environment"`
}

type Materials struct {
	FloorM2    float64 `json:"floor_m2"`
	MortarBags int     `json:"mortar_bags"`
	GroutKg    int     `json:"grout_kg"`
}

type FloorCalculationResponse struct {
	LaborCost     float64   `json:"labor_cost"`
	Materials     Materials `json:"materials"`
	EstimatedDays int       `json:"estimated_days"`
}

File: obrapro-web/src/types/calculate.ts
export type Materials = {
    floor_m2: number
    mortar_bags: number
    grout_kg: number
}

export type PaintMaterials = {
    paint_liters: number
    massa_corrida_kg: number
    fundo_liters: number
}

export type FloorCalculationResponse = {
    labor_cost: number
    material_cost: number
    total_cost: number
    materials: Materials
    estimated_days: number
}

export type PaintCalculationResponse = {
    labor_cost: number
    material_cost: number
    total_cost: number
    paint_materials: PaintMaterials
    estimated_days: number
}

export type DemolitionResponse = {
    labor_cost: number
    total_cost: number
    estimated_days: number
}

export type CalculationResult = FloorCalculationResponse | PaintCalculationResponse | DemolitionResponse

export function isFloorResult(r: CalculationResult): r is FloorCalculationResponse {
    return "materials" in r
}

export function isPaintResult(r: CalculationResult): r is PaintCalculationResponse {
    return "paint_materials" in r
}

export function isDemolitionResult(r: CalculationResult): r is DemolitionResponse {
    return !("materials" in r) && !("paint_materials" in r)
}